OCPRHV-482: Adding auto pinning policy and hugepages to RHV machine set sample YML

emarcusRH opened the pull request:
Enterprise-4.9
https://issues.redhat.com/browse/OCPRHV-482
Auto NUMA and CPU pinning, and Hugepages  added to sample YML, and to Additional RHV machine pool parameters

previews: https://deploy-preview-35574--osdocs.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-rhv?utm_source=github&utm_campaign=bot_dp
https://deploy-preview-35574--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-customizations?utm_source=github&utm_campaign=bot_dp

Gal-Zaidman commented on modules/machineset-yaml-rhv.adoc:
```diff
@@ -59,8 +59,10 @@ spec:
           type: <workload_type> <19>
           userDataSecret:
             name: worker-user-data
+            auto_pinning_policy: <auto_pinning_policy> <20>
+            hugepages: <hugepages> <21>
```
both fields needs to be one indentation less, and I think we should have the `userDataSecret` last so basically change it to:

```yaml
          type: <workload_type> <19>
          auto_pinning_policy: <auto_pinning_policy> <20>
          hugepages: <hugepages> <21>
          affinityGroupsNames:
            - compute <22>
          userDataSecret:
            name: worker-user-data
```

Gal-Zaidman commented on modules/machineset-yaml-rhv.adoc:
```diff
@@ -102,9 +104,10 @@ $ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
-<20> A list of affinity group names that should be applied to the VMs. The affinity groups must exist in oVirt.
+<20> Optional: AutoPinningPolicy defines the policy that automatically sets CPU and NUMA settings, including pinning to the host for this instance. Supported values: `none`, `resize_and_pin`.
+<21> Optional: Hugepages is the size in KiB for defining hugepages in a VM. Supported values: 2048 or 1048576.
+<22> A list of affinity group names that should be applied to the VMs. The affinity groups must exist in oVirt.
```
This is also "Optional"

Gal-Zaidman commented:
What about the installer part? there is a HUGE part missing here.
The fields need to be added to the install config table here:
https://deploy-preview-35574--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-customizations.html#installation-configuration-parameters-additional-machine_installing-rhv-customizations

emarcusRH commented:
> What about the installer part? there is a HUGE part missing here.
> The fields need to be added to the install config table here:
> https://deploy-preview-35574--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-customizations.html#installation-configuration-parameters-additional-machine_installing-rhv-customizations

Hi Gal - I wanted to handle the installer information in a separate PR, but I can add it here if you think it's also simple item to review/approve

emarcusRH commented:
@Gal-Zaidman 
1. added RHV machine pool params to the document  - https://deploy-preview-35574--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_rhv/installing-rhv-customizations?utm_source=github&utm_campaign=bot_dp#installation-configuration-parameters-additional-machine_installing-rhv-customizations
2. Updated indentation and order of params in the sample YML - https://deploy-preview-35574--osdocs.netlify.app/openshift-enterprise/latest/machine_management/creating_machinesets/creating-machineset-rhv?utm_source=github&utm_campaign=bot_dp

Gal-Zaidman commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1094,6 +1114,36 @@ This example defines no affinity groups:
+
+[source,yaml]
+----
+<machine-pool>:
+autoPinningPolicy:
+description: AutoPinningPolicy defines the policy to automatically set the CPU and NUMA including pinning to the host for the instance. When the field is omitted the default will be "none".
+  enum:
+    - none
+    - resize_and_pin
+  type: string
+----
```
All this Yaml is wrong, and it broke the table layout.
You should take a look at the layout before tagging us for a review because this looks very very bad docs-wise.

You don't need the yaml here you can just keep the description you added

Gal-Zaidman commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1094,6 +1114,36 @@ This example defines no affinity groups:
+|`<machine-pool>.platform.ovirt.hugepages`
+|Optional. Hugepages is the size in KiB for defining hugepages in a VM. Supported values: 2048 or 1048576.
+
+[source,yaml]
+----
+<machine-pool>:
+hugepages:
+  description: Hugepages is the size of a VM's hugepages to use in KiBs.
+  enum:
+    - 2048
+    - 1048576
+  format: int32
+  type: integer
+----
```
All this Yaml is wrong, and it broke the table layout.
You should take a look at the layout before tagging us for a review because this looks very very bad docs-wise.

You don't need the yaml here you can just keep the description you added

Gal-Zaidman commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1094,6 +1114,36 @@ This example defines no affinity groups:
+
+|String
+|`<machine-pool>.platform.ovirt.hugepages`
+|Optional. Hugepages is the size in KiB for defining hugepages in a VM. Supported values: 2048 or 1048576.
```
Maybe give a link to the RHV docs about hugepages? not a must, only if you think this is needed

Gal-Zaidman commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1094,6 +1114,36 @@ This example defines no affinity groups:
 ----
 |String
+|`<machine-pool>.platform.ovirt.AutoPinningPolicy`
+| Optional. AutoPinningPolicy defines the policy to automatically set the CPU and NUMA settings including pinning to the host for the instance. When the field is omitted, the default is `none`. Supported values: `none`, `resize_and_pin`.
```
Maybe give a link to the RHV docs about Auto Pinning Policy? not a must, only if you think this is needed

Gal-Zaidman commented:
There are git issues in this PR, if you look at the commits there are a bunch of things that got in here that shouldn't be here.
You need to rebase on master, I can help you with that if you want just set up a 30m call and we can take a look at git together.

emarcusRH commented:
Closing this PR and starting a new one to resolve commit errors

Gal-Zaidman commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1094,6 +1114,36 @@ This example defines no affinity groups:
+  format: int32
+  type: integer
+----
+|String
```
This is an Integer
